tests/test_copy.py

```diff
@@ -42,30 +48,36 @@ def test_add_post_data_redirect_status_code(
         "Item_description view redirected from Add_copy view wrong response"
 
 
-def test_db_after_add_copy(copy_form, db_book, client,
-                           login_form_admin_credentials):
-    client.post(url_for('library.login'),
-                data=login_form_admin_credentials.data)
-    client.post(url_for('library.add_copy',
-                        item_id=db_book.id),
-                data=copy_form[0].data)
-    copy = Copy.query.filter_by(
-        asset_code=copy_form[0].asset_code.data).first()
-    assert copy, \
-        "Copy not added to db"
-    assert copy.library_item_id == db_book.id, \
-        "Added copy pointed wrong book id"
-    assert copy.library_item == db_book, \
-        "Added copy pointed wrong book"
-    assert copy.available_status, \
-        "Added copy has wrong available status"
-    assert copy.has_cd_disk, \
-        "Added copy has wrong has_cd_disk status"
-    assert copy.asset_code == copy_form[0].asset_code.data, \
-        "Added copy asset_code is not the one from the Form"
-    assert copy.shelf == copy_form[0].shelf.data, \
-        "Added copy shelf is not the one from the Form"
-    session.clear()
+def test_db_after_add_copy(
+    copy_form,
+    db_book,
+    client,
+    login_form_admin_credentials,
+    mock_ldap
+):
+    with mock.patch('views.index.ldap_client', mock_ldap):
+        client.post(url_for('library.login'),
+                    data=login_form_admin_credentials.data)
+        client.post(url_for('library.add_copy',
+                            item_id=db_book.id),
+                    data=copy_form[0].data)
+        copy = Copy.query.filter_by(
+            asset_code=copy_form[0].asset_code.data).first()
+        assert copy, \
```

Comment:
A good habit of writing tests is writing your own assertions. Try write own assertion eg. `assert_copies(copy_1, copy_2)`